docs/architecture.md

```diff
@@ -0,0 +1,147 @@
+whole bare metal machine as the the Kubernetes node and inside of that we can have 1 or more TEE VMs, in which we
+can run our pod sandboxes.
+Although this is the traditional way to currently configure a Kata Containers VMs in the cloud, there is a big
+drawback of having to have a separate bare metal box per Kubernetes node, which is expensive and difficult to scale.
+
+
+### Nested Virtualisation
+
+Most current cloud providers need to have the scalabilty and cost savings that bare metal nodes cannot provide, so
+utilise virtual servers. These partition up the physical processing power and resources into multiple systems that
+can each run their own operating system. A common model in non-confidential Kubernetes infrastructure is to have a
+Kubernetes node map to a virtual server, so each physical server can run multiple nodes and each of them could run
+multiple pod sandboxes. Currently there are limitation of not being able to run as a second-level VM in a performant
+way for most TEEs and more CSP, means we can't put these pod sandboxes inside a TEE boundary in these scenarios.
+
+
+### Peer pods
+
+The peer pod approach is to compromise between the two previous models - using a TEE environment as a first-level KVM,
+but not having to dedicate a whole machine to a single Kubernetes node and therefore getting the flexibility of
+virtualisation.
+In this model the pod sandbox is not within the Kubernetes node, using some of its resources, but an independent VM
+that runs in the same cloud as a first-level KVM peer of the Kubernetes node. CSP IaaS APIs can be used to create these
+new TEE-based VSIs on demand for the pod sandboxes and we can create a network tunnel between the node and sandbox for
+commands to flow.
+
+There are some limitations and advantages of this approach including:
+
+#### Limitations
+
+In the current Kata Containers approach, the container runtime creates a pod sandbox as a VM inside the Kubernetes node
+using a local hypervisor and then uses a local virtual network to communicate to it. Compared to this existing approach,
+using CSP IaaS APIs to create a new VM per pod sandbox adds an overhead in time and resources. This means the peer pods
+solution probably won't work well for small, short-lived applications.
+Using a virtual network to communicate between the node (host) and the pod sandbox (guest) components, the approach of
+peer pods is more complicated as well
+
+#### Advantages
+Due to the lack of wide availability of running TEEs in nested VM, if we want to use TEEs, then compared to having a bare
+metal machine per node, we can scale more easily and cheaply.
+There is also some advantages to the independence of peer pods, where each pod sandbox has it's own dedicated resources
+like network and storage, rather than sharing these with the Kubernetes node and other pods. This can help to improve
+resource management, reliability as well as security.
```

Comment:
Another potential advantage is that we don't require any changes to the Cloud API